instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt

```diff
@@ -0,0 +1,154 @@
+package org.oppia.android.instrumentation.player
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByDesc
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByText
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.scrollRecyclerViewTextIntoView
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.startOppiaFromScratch
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForRes
+
+/** Tests for Explorations. */
+class ExplorationPlayerTest {
+  private lateinit var device: UiDevice
+
+  @Before
+  fun setUp() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.startOppiaFromScratch()
+  }
+
+  @Test
+  fun testExploration_prototypeExploration_toolbarTitle_isDisplayedSuccessfully() {
+    navigateToPrototypeExploration()
+    assertThat(device.findObjectByRes("exploration_toolbar_title")).isNotNull()
+  }
+
+  @Test
+  fun testProtoTypeExploration_answerAllInteractionsAndCompleteExploration() {
+    navigateToPrototypeExploration()
+    completePrototypeExploration()
```

Comment:
I think this is hiding a bit too much from the test. Could you instead introduce multiple helper methods (similar to StateFragmentLocalTest) for each part of the flow, and then expand the actual completion bits in the test so that context isn't lost?

Reply:
Done

Reply:
To be a bit more clear: this test should go through all the steps because they are key actions for the test itself. The next test can just rely on the helper since it's part of the arrangement of that test rather than the actual action.